### PR TITLE
Consolidate modelm v2

### PR DESCRIPTION
**Pull request by nuess0r**
As discussed in #12370:
The maintainers of several Model M replacement
controller projects agreed to consolidate them
under one common folder.

This pull request replaces #12792 as I was unable to undo the merge instead of rebase that I did there.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* #12370

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by tzarc** [comment]
Keyboard deletes and moves need to occur on a breaking changes cycle, so that it can be properly documented for affected users.

This means the PR has been retargeted against the `develop` branch.

Unfortunately this has conflicts (and did so before when it targeted `master`) -- please fix up the conflicts and create a list of the consolidations so that the appropriate breaking changes documentation can be created.

**Comment by nuess0r** [comment]
> Keyboard deletes and moves need to occur on a breaking changes cycle, so that it can be properly documented for affected users.

Hi

I tried already three times to get this pull-request in and always in the meantime until it was reviewed it got conflicts.

To make our lives easier I like to ask if you have something like a merge window or freeze before releasing a new version?
Could you suggest me some specific week (or a range of few days) where I shall rebase my changes again?

**Comment by fauxpark** [comment]
It's in the docs: https://docs.qmk.fm/#/breaking_changes?id=important-dates

**Comment by nuess0r** [comment]
Am Tue, 12 Jul 2022 19:17:33 -0700
schrieb "github-actions[bot]" ***@***.***>:

> Thank you for your contribution!
> This pull request has been automatically marked as stale because it
> has not had activity in the last 45 days. It will be closed in 30
> days if no further activity occurs. Please feel free to give a status
> update now, or re-open when it's ready. For maintainers: Please label
> with `bug`, `awaiting review`, `breaking_change`, `in progress`, or
> `on hold` to prevent the issue from being re-flagged.
> 

I missed the last breaking changes release cycle but looking forward
for the upcoming one. I'm following your wiki page about the release
cycle, so I know when to do another rebase.

**Comment by nuess0r** [comment]
List of the consolidations:

- ashpil_usbc 
- ibm122m
- mschwingen
- teensy2 (renamed from modelm101_teensy2)
- teensypp (renamed from modelm101)
- teensypp_ssk (renamed from modelm_ssk)
- yugo_m

**Review by tzarc — APPROVED** [review]

File: keyboards/modelm/ibm122m/readme.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# ibm122m
+
+![IBM Model M 122 key](https://i.imgur.com/Oo3Ozqz.jpg)
+
+This is a keymap for the IBM Model M 122 key terminal keyboard running on a Teensy++ 2.0
+I wired it to weird pins on mine (mainly to accomodate two speakers), so make sure to update the pin arrays.
+
+Keyboard Maintainer: [Luke Stanley](https://github.com/lukexorz)
+Hardware Supported: Teensy++ 2.0
+Hardware Availability: https://www.pjrc.com/store/teensypp.html
```
Comment (fauxpark) [inline keyboards/modelm/ibm122m/readme.md:10]:
```suggestion
* Keyboard Maintainer: [Luke Stanley](https://github.com/lukexorz)
* Hardware Supported: Teensy++ 2.0
* Hardware Availability: https://www.pjrc.com/store/teensypp.html
```

File: keyboards/modelm/mschwingen/led_ffc/readme.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+* Keyboard Maintainer: [Michael Schwingen](https://github.com/mschwingen/)
+* Hardware Supported: [Model M USB PCB](https://github.com/mschwingen/hardware/tree/master/modelm-usb)
+* Hardware Availability: need to build your own.
+
+Make example for this keyboard (after setting up your build environment), run one of:
+
+    make modelm/mschwingen/led_ffc:default
+
+flash:
```
Comment (fauxpark) [inline keyboards/modelm/mschwingen/led_ffc/readme.md:15]:
```suggestion
Flashing example for this keyboard:
```

File: keyboards/modelm/mschwingen/led_wired/readme.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+* Keyboard Maintainer: [Michael Schwingen](https://github.com/mschwingen/)
+* Hardware Supported: [Model M USB PCB](https://github.com/mschwingen/hardware/tree/master/modelm-usb)
+* Hardware Availability: need to build your own.
+
+Make example for this keyboard (after setting up your build environment), run one of:
+
+    make modelm/mschwingen/led_wired:default
+
+flash:
```
Comment (fauxpark) [inline keyboards/modelm/mschwingen/led_wired/readme.md:15]:
```suggestion
Flashing example for this keyboard:
```

File: keyboards/modelm/mschwingen/led_ws2812/readme.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+* Keyboard Maintainer: [Michael Schwingen](https://github.com/mschwingen/)
+* Hardware Supported: [Model M USB PCB](https://github.com/mschwingen/hardware/tree/master/modelm-usb)
+* Hardware Availability: need to build your own.
+
+Make example for this keyboard (after setting up your build environment), run one of:
+
+    make modelm/mschwingen/led_ws2812:default
+
+flash:
```
Comment (fauxpark) [inline keyboards/modelm/mschwingen/led_ws2812/readme.md:15]:
```suggestion
Flashing example for this keyboard:
```

File: keyboards/modelm/teensypp/readme.md [hunk 5]
```diff
@@ -27,16 +27,16 @@ Pins: F7 F6 F5 F4 F3 F2 F1 F0
...
+```
 
-Please note that some versions of Model M have LED connected with a ribbon cable, and not with separate wires.  
+Please note that some versions of Model M have LED connected with a ribbon cable, and not with separate wires.
 
-Keyboard Maintainer: [iw0rm3r](https://github.com/iw0rm3r)  
-Hardware Supported: Teensy 2.0++ board by PJRC  
+Keyboard Maintainer: [iw0rm3r](https://github.com/iw0rm3r)
+Hardware Supported: Teensy++ 2.0 board by PJRC
 Hardware Availability: https://www.pjrc.com/store/teensypp.html
```
Comment (fauxpark) [inline keyboards/modelm/teensypp/readme.md:36]:
```suggestion
* Keyboard Maintainer: [iw0rm3r](https://github.com/iw0rm3r)
* Hardware Supported: Teensy++ 2.0 board by PJRC
* Hardware Availability: https://www.pjrc.com/store/teensypp.html 
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
.noci can also be deleted.

**Comment by nuess0r** [comment]
@tzarc What was blocking this pull request to be merged in breaking_change_2023q1?

**Comment by nuess0r** [comment]
Updated everything for the next upcoming breaking changes cycle. Please review.

File: keyboards/kprepublic/bm80v2/keymaps/smooted/config.h
Comment (keyboard-magpie) [inline keyboards/kprepublic/bm80v2/keymaps/smooted/config.h:1]: Not sure why this has made it in here?

File: keyboards/lxxt/config.h
Comment (keyboard-magpie) [inline keyboards/lxxt/config.h:1]: Same here. Not sure this is an intended inclusion.

File: keyboards/kprepublic/bm80v2/keymaps/smooted/config.h
Comment (nuess0r) [reply keyboards/kprepublic/bm80v2/keymaps/smooted/config.h:1]: Thanks for finding this. I missed it somehow and fixed it.

These files are touched due to the recent changes towards more data driven keyboard definitions. At some point lot of files got removed from the repository. Doing a rebase then with a patch set that moves lot of files made Git a bit confused and with few files Git tough a Model M related file was moved to somewhere completely different.

File: keyboards/lxxt/config.h
Comment (nuess0r) [reply keyboards/lxxt/config.h:1]: This file remains the same as in the current repository:
https://github.com/qmk/qmk_firmware/blob/master/keyboards/lxxt/config.h

These files are touched due to the recent changes towards more data driven keyboard definitions. At some point lot of files got removed from the repository. Doing a rebase then with a patch set that moves lot of files made Git a bit confused and with few files Git tough a Model M related file was moved to somewhere completely different.

**Review by tzarc — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by tzarc** [comment]
CI build failure internal logs throw up errors with a single word: `matrix_size`. Investigating.

**Review by tzarc — APPROVED** [review]

**Pull request merged**